### ISSUE TITLE
Add photos to contact details of contributors

### ISSUE THREAD

**Issue by FrauCsu** [tactilenews/100eyes#226]
As editor I want to have to possibility
to manually add pictures to contributor's profiles.
E.g. when they send pics of themselves I want to be able to use them as avatars

**Comment by tillprochaska** [comment]
That’s a duplicate of #213?

**Comment by FrauCsu** [comment]
I want to use pictures contributors send me via 100eyes as profile pics. If that's a duplicate, then close issue 226

**Comment by phoeinx** [comment]
So if I get it right #213 would refer to being able to change the profile pic when editing the information of a user, e.g. by clicking on the avatar and then being able to upload a picture? 

And #226 would be literally a direct link from a picture a user sent for setting this picture as the profile pic. 

Is both wanted or only #226 for now?

In regards to UI for #226 - the easiest way would be to just add a button in the meta option line of each chat message containing a photo. (With a different icon of course)
![image](https://user-images.githubusercontent.com/34538290/96510684-31672080-125e-11eb-8404-5437b70bbb96.png)

Would this work?

I feel that it makes the message box appear very broad - and the meta-information somehow nearly bigger than the actual content. A second option would be to always place the "Add as profile pic" option in a row below the other options. (This right now happens automatically when the width of the screen is reduced.)
![image](https://user-images.githubusercontent.com/34538290/96511199-09c48800-125f-11eb-9b26-bf047d925449.png)

**Comment by tillprochaska** [comment]
@phoeinx I think the main problem with this solution is that a message may contain more than one photo, i.e. I guess the "use as profile picture" action would need to be tied to the attached images, not the message.

I could imagine two possible solutions for this:

* Clicking the image thumbnail doesn’t directly open the full-size photo, but instead shows a dropdown that allows you to download/open full-size image or use that image as a profile picture.
* Clicking the images doesn’t simply open the full-size image file in a new tab, but instead shows the photo in an overlay along with actions to download or use as a profile picture. (I think we discussed that previously and it’s something we wanted to implement eventually
* Any other ideas?

I might add two quick sketches this afternoon to make this clearer.

**Comment by tillprochaska** [comment]
<img width="712" alt="Screen Shot 2020-10-20 at 11 39 50" src="https://user-images.githubusercontent.com/1512805/96569088-fcdc7e80-12c8-11eb-85ec-65b8f839eb3c.png">

**Comment by tillprochaska** [comment]
> So if I get it right #213 would refer to being able to change the profile pic when editing the information of a user, e.g. by clicking on the avatar and then being able to upload a picture?
> 
> And #226 would be literally a direct link from a picture a user sent for setting this picture as the profile pic.

Yes, that’s the difference!

> Is both wanted or only #226 for now?

I think using a picture from a contributor’s reply (i.e. #226) ist most important right now, but let’s double-check that with @FrauCsu and @drjakob.

**Comment by tillprochaska** [comment]
> Clicking the image thumbnail doesn’t directly open the full-size photo, but instead shows a dropdown that allows you to download/open full-size image or use that image as a profile picture.

<img width="612" alt="Screen Shot 2020-10-20 at 14 54 05" src="https://user-images.githubusercontent.com/1512805/96588582-1fc85c00-12e4-11eb-89c1-67e1c77601dd.png">

> Clicking the images doesn’t simply open the full-size image file in a new tab, but instead shows the photo in an overlay along with actions to download or use as a profile picture. (I think we discussed that previously and it’s something we wanted to implement eventually

Similar to the [media viewer in Wikipedia](https://de.wikipedia.org/wiki/Golden_Retriever#/media/Datei:Golden_Retriever_Carlos_(10581910556).jpg). But I’d probably go for one of the other options (as they probably are way easier to implement), if there aren’t strong reasons for this one.

<img width="790" alt="Screen Shot 2020-10-20 at 15 06 56" src="https://user-images.githubusercontent.com/1512805/96590133-ea247280-12e5-11eb-9e06-2e1a68c09dfc.png">

Let me know when you’d like to discuss UI stuff or need help!

**Comment by FrauCsu** [comment]
@phoeinx @tillprochaska : Issues #213 and #226 seem to be duplicates. As editor I want to be able to do three things regarding photos in 100eyes:

a) download photos that were send by contributors and use them on my other platforms (e.g. wordpress)
b) download a photo that was send by a contributor and use it as their profile picture
c) send photos from my other platforms to contributors (e.g. a picture of my newsroom staff)

In order to do (b) I need the possibility to add a picture to a person's profile in the profile section (like on facebook, where there is this little camera symbol next to the profile picture:
<img width="576" alt="Bildschirmfoto 2020-10-20 um 16 06 51" src="https://user-images.githubusercontent.com/44811680/96597587-4ab7ad80-12ee-11eb-81b2-6aec973074cc.png">

**Comment by tillprochaska** [comment]
> b) download a photo that was send by a contributor and use it as their profile picture

Why do the extra work and download the photo, then reupload it, if you could set the profile picture with a single click?

**Comment by FrauCsu** [comment]
@tillprochaska If that means I will be asked every time a user sends a pic whether I want to use it as profile pic or download it, I prefer just the download option. The profile pic is a one time thing, the download option is needed whenever a user sends a picture.

**Comment by phoeinx** [comment]
hi everyone,

Thanks for the reply @tillprochaska :)
Regarding multiple pictures I figured we could place a "set as profile picture" button below each picture, but that would clutter the UI a lot. I like your ideas way better.
Nice choice of pictures btw. 

@FrauCsu good to know.
a) is already possible by downloading the image from the tab where it opens when clicked. 
<img width="1438" alt="image" src="https://user-images.githubusercontent.com/34538290/96651073-5332d700-1334-11eb-9b2a-e68354a0931a.png">
Is that good enough or would you prefer some of the prettier solutions (dropdown, overlay) Till proposed above?
b) alright, I'll work on that as soon as I have time (and close #213 as it now officially is a duplicate)
c) would be issue #383, right?

**Comment by tillprochaska** [comment]
> @tillprochaska If that means I will be asked every time a user sends a pic whether I want to use it as profile pic or download it, I prefer just the download option. The profile pic is a one time thing, the download option is needed whenever a user sends a picture

@FrauCsu What you can already do right now wouldn’t change as @phoeinx pointed out. You’d still be able to click the thumbnail to see a large version or save it. But you could also click the menu icon to see additional options.

Out of curiosity: You’re downloading every single picture a contributor sends via 100eyes directly when they come in? What are you doing with the downloaded files? Why?

**Comment by FrauCsu** [comment]
No, misunderstanding, @tillprochaska : I am not downloading all of them - only the ones I want to use for my report.

**Comment by FrauCsu** [comment]
@tillprochaska I just like the little camera button 
<img width="576" alt="Bildschirmfoto 2020-10-20 um 16 06 51" src="https://user-images.githubusercontent.com/44811680/96734031-66888580-13ba-11eb-853a-302fdffb1e8c.png">

**Comment by FrauCsu** [comment]
@phoeinx @tillprochaska 
a) good enough
b) @phoeinx next job
c) #383, yes.
